### PR TITLE
Allow spaces for codespan containing backticks (MD038)

### PR DESCRIPTION
**Pull request by jakeprime**
## Description

Fixes edge case in rule MD038 where if the codespan was within parentheses the surrounding spaces rule could be incorrectly raise. For example:

```
Use 3 backticks (` ``` `)
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (non-breaking change that does not add functionality but updates documentation)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/markdownlint/markdownlint/blob/main/CONTRIBUTING.md) document.
- [x] Wrote [good commit messages](https://chris.beams.io/posts/git-commit/)
- [x] Feature branch is up-to-date with `main`, if not - rebase it
- [x] Added tests for all new/changed functionality, including tests for positive and negative scenarios
- [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences

### CODE REVIEW

File: test/rule_tests/spaces_inside_codespan_elements.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 `normal codespan element`
 
+Spaces are necessary when the contents of the codespan itself starts or ends
+with a backtick
+`` `Hello, ${name}` ``
```
Comment (jaymzh) [inline test/rule_tests/spaces_inside_codespan_elements.md:5]: I'm not sure this is valid - codeblocks are 3 backticks, not 2.

Comment (jakeprime) [reply test/rule_tests/spaces_inside_codespan_elements.md:5]: Thank for reviewing @jaymzh 🙏🏻

Code _blocks_ are delineated by 3 (or more) backticks on a single line. This rule is for inline code _spans_. In that case they are usually a single backtick, but can be any number as long as the open and close match. If the contents contains a backtick then you'd use more than one to prevent it closing the span.

There are some good examples [here](https://daringfireball.net/projects/markdown/syntax#code).

File: test/rule_tests/spaces_inside_codespan_elements.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+` ``` `
```
Comment (jaymzh) [inline test/rule_tests/spaces_inside_codespan_elements.md]: This test passes with and without your change, so not a great test.

I think that's because kramdown will strip these when they are semetrical (one space on either side). You need to add a test that fails in the current world. ;)

**Review by jaymzh — CHANGES_REQUESTED** [review]
Oh that makes sense. Bug in your test though.

**Comment by jaymzh** [comment]
ping @jakeprime

**Comment by jakeprime** [comment]
@jaymzh Ah yes, not sure how I didn't spot that 🙃 

This looks like an existing issue that already acknowledged with the other examples in that test file:

https://github.com/markdownlint/markdownlint/blob/03ff44490b0631c3c8e756f008d3d2af41852592/test/rule_tests/spaces_inside_codespan_elements.md?plain=1#L5-L8

Do you think that means it's OK as it is? I'll try to find some time to look into it, but at first sight I think this means updating the custom Kramdown parser. If it currently fails to parse any codespan that starts with a space then I can't create any failing examples.

**Comment by jaymzh** [comment]
@jakeprime  I'm not asking you to fix it, I'm asking you to update your test to show the actual problem you ran into (add a space on one side or the other)

**Comment by jakeprime** [comment]
@jaymzh Ah sorry, yes I get you now. This turned out to be more edge-casey than I originally assumed. It's not the surrounding spaces themselves that are the issue, it's if the codespan is surrounded by parentheses such as in the lint failure that brought me here in the first place which was:

```
- 3 backticks (` ``` `) on its own line before and after the block
```

Or more likely it's having any character adjacent to the delimiting backticks, but I've only tested with parentheses. But whatever I now have a previously failing test example.

**Comment by jaymzh** [comment]
I asked claude to try a bunch of cases and it found that brackets and quotes also trigger the same bug that you fixed. You wanna add those to your test case?

**Comment by jakeprime** [comment]
@jaymzh Yeah sure thing. I've added a couple of variations.

**Review by jaymzh — APPROVED** [review]

**Comment by jaymzh** [comment]
Thanks!
